src/formatting_layer.rs

```diff
@@ -86,9 +87,23 @@ impl<W: for<'a> MakeWriter<'a> + 'static> BunyanFormattingLayer<W> {
             hostname: gethostname::gethostname().to_string_lossy().into_owned(),
             bunyan_version: 0,
             default_fields,
+            skip_fields: HashSet::new(),
         }
     }
 
+    /// Add fields to skip when formatting with this layer.
+    pub fn skip_fields<T>(mut self, fields: &[T]) -> Self
+    where
+        T: AsRef<str>,
+    {
+        for field in fields {
+            let field = field.as_ref().to_string();
+            self.skip_fields.insert(field);
+        }
```

Comment:
We should return an error if they specify a bunyan core field in the skip list.

Reply:
I can do that but with the core fields defined right now that would mean I can't skip `file` and `line` which are the main fields I want to remove from my output.

I took a quick look at the [core fields](https://github.com/trentm/node-bunyan#core-fields) and apparently the source info under `src` is optional, would you be ok with making `file` and `line` non-reserved ?

Reply:
My bad, I should have been more precise: all non-optional core fields.

Reply:
oh right.